core/domain/user_services.py

```diff
@@ -1729,3 +1798,203 @@ def is_topic_manager(user_id):
+        language_code: str. The code of the language.
+
+    Returns:
+        bool. Whether the user can review voiceover applications in any language
+        or in the given language.
+    """
+    user_community_rights = get_user_community_rights(user_id)
+    reviewable_language_codes = (
+        user_community_rights.can_review_voiceover_for_language_codes)
+    if language_code is not None:
+        return language_code in reviewable_language_codes
+    else:
+        return bool(reviewable_language_codes)
+
+
+def can_review_question_suggestions(user_id):
+    """Checks whether the user can review question suggestions.
+
+    Args:
+        user_id: str. The unique ID of the user.
+
+    Returns:
+        bool. Whether the user can review question suggestions.
+    """
+    user_community_rights = get_user_community_rights(user_id)
+    return user_community_rights.can_review_questions
+
+
+def allow_user_to_review_translation_in_language(user_id, language_code):
+    """Allows the user with the given user id to review translation in the given
+    language_code.
+
+    Args:
+        user_id: str. The unique ID of the user.
+        language_code: str. The code of the language. Callers should ensure that
+            the user does not have rights to review translations in the given
+            language code.
+    """
+    user_community_rights = get_user_community_rights(user_id)
+    language_set = set(
+        user_community_rights.can_review_translation_for_language_codes)
+    language_set.add(language_code)
+    sorted_language_list = sorted(list(language_set))
+    user_community_rights.can_review_translation_for_language_codes = (
+        sorted_language_list)
+    _save_user_community_rights(user_community_rights)
+
+
+def remove_translation_review_rights_in_language(user_id, language_code):
+    """Removes the user's review rights to translation suggestions in the given
+    language_code.
+
+    Args:
+        user_id: str. The unique ID of the user.
+        language_code: str. The code of the language. Callers should ensure that
+            the user have rights to review translations in the given
+            language code.
+    """
+    user_community_rights = get_user_community_rights(user_id)
+    user_community_rights.can_review_translation_for_language_codes.remove(
+        language_code)
+    _update_user_community_rights(user_community_rights)
+
+
+def allow_user_to_review_voiceover_in_language(user_id, language_code):
+    """Allows the user with the given user id to review voiceover applications
+    in the given language_code.
+
+    Args:
+        user_id: str. The unique ID of the user.
+        language_code: str. The code of the language. Callers should ensure that
+            the user does not have rights to review voiceovers in the given
+            language code.
+    """
+    user_community_rights = get_user_community_rights(user_id)
+    language_set = set(
```

Comment:
Same thing, throughout this function, and elsewhere: distinguish between languages and language codes.